### PR TITLE
feat: adding init command to entrypoints

### PR DESCRIPTION
**Pull request by gvauter**
## Description

This is the initial pull request for creating a new `init` command in trestlebot.  This entrypoint is used to initialize an empty Git repository for use with trestlebot.  This PR only includes support for GitHub as the CICD provider.  Future PRs will expand the support to other providers.

Fixes #[321](https://github.com/RedHatProductSecurity/trestle-bot/issues/321)

## Type of change

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

## How has this been tested?

- [X] Created new testing module for this entrypoint
- [X] Tested new entrypoint locally

**Test Configuration**:

- OS: Fedora 39
- Python: 3.11.9

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation # Will come in a future PR
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: trestlebot/entrypoints/init.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+
+logger = logging.getLogger(__name__)
+logging.getLogger("trestle.core.commands.init").setLevel("CRITICAL")
+
+OSCAL_MODEL_SSP = model_types.AuthoredType.SSP.value
+OSCAL_MODEL_COMPDEF = model_types.AuthoredType.COMPDEF.value
+
+
+class InitEntrypoint(EntrypointBase):
```
Comment (jpower432) [inline trestlebot/entrypoints/init.py:48]: Wondering if this should not inherit from EntrypointBase and if we should put this command in a different location given that it does not share the same core logic as the others. I think the only flags `init` would need to duplicate is `verbose` and `workspace`. The `dry-run` flag does not appear to be used.

File: trestlebot/entrypoints/templates/github/trestlebot-rules-transform.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+---
```
Comment (jpower432) [inline trestlebot/entrypoints/templates/github/trestlebot-rules-transform.yml:1]: Take a look at this updates version -> https://github.com/jpower432/trestlebot-demo-cd/blob/main/.github/workflows/ci.yml

**Review by jpower432 — CHANGES_REQUESTED** [review]
Couple initial questions and feedback

File: trestlebot/entrypoints/init.py (hunk 1 above)
Comment (gvauter) [reply trestlebot/entrypoints/init.py:48]: Yes, that sounds good to me.

File: trestlebot/entrypoints/templates/github/trestlebot-rules-transform.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+name: Trestle-bot rules-transform and autosync
+
+on:
+  pull_request:
```
Comment (jpower432) [inline trestlebot/entrypoints/templates/github/trestlebot-rules-transform.yml:5]:
```suggestion
  push:
```
I think we could use `dry_run` on pull request based workflow and `push` by default. The goal here to reduce merge conflict and review of generated content while still being able to verify it before merge. WDYT? This we would be the same change for all the sync-based templates.

**Review by jpower432 — COMMENTED** [review]
Code changes look good. Question on the templates only.

File: trestlebot/entrypoints/init.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    def _call_trestle_init(self, args: argparse.Namespace) -> None:
+        """Call compliance-trestle to initialize workspace"""
+        trestle_args = argparse.Namespace(
+            verbose=args.verbose,
+            trestle_root=pathlib.Path(args.working_dir),
+            full=False,
+            govdocs=True,
+            local=False,
+        )
+        InitCmd()._run(trestle_args)
```
Comment (jpower432) [inline trestlebot/entrypoints/init.py:131]: Might be worth checking the error code here. `trestle` store that information [here](https://github.com/oscal-compass/compliance-trestle/blob/6c2d3f3bd8d6eeaf04e0a931ce39b8b52646e95a/trestle/core/commands/common/return_codes.py#L36).

**Review by jpower432 — APPROVED** [review]
LGTM

**Pull request merged**